From e4bf129ab6e34818981f44aee5f6d44dfc248147 Mon Sep 17 00:00:00 2001
From: zhangbo <zbsdta@126.com>
Date: Sat, 13 Jan 2018 20:03:33 +0800
Subject: [PATCH] add power on alarm debug log

---
 fs/timerfd.c             | 4 ++++
 kernel/time/alarmtimer.c | 2 ++
 2 files changed, 6 insertions(+)

diff --git a/fs/timerfd.c b/fs/timerfd.c
index 31374ec8f9bd..d623ef1f0a5b 100644
--- a/fs/timerfd.c
+++ b/fs/timerfd.c
@@ -488,6 +488,10 @@ static int do_timerfd_settime(int ufd, int flags,
 
 	spin_unlock_irq(&ctx->wqh.lock);
 
+	if (new->it_value.tv_sec > 5 || new->it_interval.tv_sec > 5)
+		pr_info("%s: clockid = %d it_value.tv_sec = %ld it_interval.tv_sec = %ld\n",
+			__func__, ctx->clockid, new->it_value.tv_sec, new->it_interval.tv_sec);
+
 	if (ctx->clockid == CLOCK_POWEROFF_ALARM)
 		set_power_on_alarm();
 
diff --git a/kernel/time/alarmtimer.c b/kernel/time/alarmtimer.c
index 47d256ff51ec..a913ef7f31f4 100644
--- a/kernel/time/alarmtimer.c
+++ b/kernel/time/alarmtimer.c
@@ -147,6 +147,8 @@ void set_power_on_alarm(void)
 	alarm_delta = wall_time.tv_sec - rtc_secs;
 	alarm_time = alarm_secs - alarm_delta;
 
+	pr_info("%s: alarm_delta = %ld\n", __func__, alarm_delta);
+
 	rtc_time_to_tm(alarm_time, &alarm.time);
 	alarm.enabled = 1;
 	rc = rtc_set_alarm(rtcdev, &alarm);
-- 
2.14.3

